### DISCUSSION TITLE
Question: Using WaveSurfer for displaying stream of Mic

### DISCUSSION THREAD

**Discussion by fatemi2020** [katspaugh/wavesurfer.js#3015] [Q&A]
Hi, thank you for sharing your great library and plugins.
I wanted to display a visualization for indicating the stream from Mic, I used WaveSurfer version 6.6.3 with the `microphone` plugin by calling the `gotStream` function, and it's working.

But you removed the `microphone` plugin and informed to use the `Record` plugin and I don't know how to use this plugin for display stream visualization.

**Comment by katspaugh** [comment]
There's an example on the website: https://wavesurfer-js.org/examples/#record.js

**Reply by fatemi2020** [reply]
I saw that example, but I don't want to record it.
I wish to display visualization waves for a stream with a voice track.

**Reply by katspaugh** [reply]
You could just start recording but ignore/discard the recorded audio.

**Reply by fatemi2020** [reply]
Actually, I'm using visualization for the voice call, and the recording will be started/stopped by the user's decision, we should start and stop recording when the user wants and should display visualization all time while the voice call is established.

**Reply by katspaugh** [reply]
Added an issue for this: https://github.com/katspaugh/wavesurfer.js/issues/3017
